test/CadlRanchProjectsNonAzure.Tests/type-property-nullable.cs

```diff
@@ -0,0 +1,248 @@
+            var response = await new NullableClient(host, null).GetDurationClient().PatchNullAsync(BinaryContent.Create(BinaryData.FromString(value)));
+            Assert.AreEqual(204, response.GetRawResponse().Status);
+        });
+
+        [Test]
+        public Task Type_Property_Nullable_CollectionsByte_getNonNull() => Test(async (host) =>
+        {
+            var response = await new NullableClient(host, null).GetCollectionsByteClient().GetNonNullAsync();
+            Assert.AreEqual("foo", response.Value.RequiredProperty);
+            BinaryDataAssert.AreEqual(BinaryData.FromString("hello, world!"), response.Value.NullableProperty.First());
+            BinaryDataAssert.AreEqual(BinaryData.FromString("hello, world!"), response.Value.NullableProperty.Last());
+            Assert.AreEqual(2, response.Value.NullableProperty.Count());
+        });
+
+        [Test]
+        public Task Type_Property_Nullable_CollectionsByte_getNull() => Test(async (host) =>
+        {
+            var response = await new NullableClient(host, null).GetCollectionsByteClient().GetNullAsync();
+            Assert.AreEqual("foo", response.Value.RequiredProperty);
+            // we will never construct a null collection therefore this property is actually undefined here.
+            Optional.IsCollectionDefined(response.Value.NullableProperty);
+        });
+
+        [Test]
+        public Task Type_Property_Nullable_CollectionsByte_patchNonNull() => Test(async (host) =>
+        {
+            var value = new
+            {
+                requiredProperty = "foo",
+                nullableProperty = new[] { "aGVsbG8sIHdvcmxkIQ==", "aGVsbG8sIHdvcmxkIQ==" }
+            };
+            var response = await new NullableClient(host, null).GetCollectionsByteClient().PatchNonNullAsync(BinaryContent.Create(BinaryData.FromObjectAsJson(value)));
+            Assert.AreEqual(204, response.GetRawResponse().Status);
+        });
+
+        [Test]
+        public Task Type_Property_Nullable_CollectionsByte_patchNull() => Test(async (host) =>
+        {
+            string value = "{ \"requiredProperty\": \"foo\", \"nullableProperty\": null }";
+            var response = await new NullableClient(host, null).GetCollectionsByteClient().PatchNullAsync(BinaryContent.Create(BinaryData.FromString(value)));
+            Assert.AreEqual(204, response.GetRawResponse().Status);
+        });
+
+        [Test]
+        public Task Type_Property_Nullable_CollectionsModel_getNonNull() => Test(async (host) =>
+        {
+            var response = await new NullableClient(host, null).GetCollectionsModelClient().GetNonNullAsync();
+            Assert.AreEqual("foo", response.Value.RequiredProperty);
+            Assert.AreEqual("hello", response.Value.NullableProperty.First().Property);
+            Assert.AreEqual("world", response.Value.NullableProperty.Last().Property);
+            Assert.AreEqual(2, response.Value.NullableProperty.Count());
```

Comment:
we could use `Count` property because this should be a list.

Reply:
fixed